control/control.go

```diff
@@ -225,6 +239,83 @@ func New(cfg *Config) *pluginControl {
 	return c
 }
 
+func (p *pluginControl) HandleGomitEvent(e gomit.Event) {
+	switch v := e.Body.(type) {
+	case *control_event.LoadPluginEvent:
+		// Get all known tasks data
+		for taskID, taskData := range p.taskIDData {
+			// Expand every task metric namespaces (with "*") to corresponding metric calalog namespaces
+			for _, tm := range taskData.metrics {
+
+				matchedNss, err := p.metricCatalog.MatchNamespaces(tm.Namespace())
+
+				if err != nil {
+					log.WithFields(log.Fields{
+						"_block":         "control",
+						"event":          v.Namespace(),
+						"plugin-name":    v.Name,
+						"plugin-version": v.Version,
+						"plugin-type":    core.PluginType(v.Type).String(),
+					}).Error("error matching task namespace with metric catalog")
+				} else {
```

Comment:
Consider using here `continue` instead of nesting the next things in _else_. 

See similar case in [/control/runner.go#L345](https://github.com/intelsdi-x/snap/blob/master/control/runner.go#L345)